gtaintegration/gtaintegration_test.go

```diff
@@ -427,6 +427,132 @@ func TestPackageRemoval_MovePackage(t *testing.T) {
 	}
 }
 
+func TestChangesAfterMergeBaseBranch(t *testing.T) {
+	ctx := context.Background()
+
+	baseBranch := t.Name() + "-base"
+	// create a base branch
+	if _, err := runGit(ctx, ".", "checkout", "-b", baseBranch, "master"); err != nil {
+		t.Fatal(err)
+	}
+
+	// move some files to a different directory
+	if _, err := runGit(ctx, ".", "mv", "src/gtaintegration/movedfrom", "src/gtaintegration/movedto"); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := runGit(ctx, ".", "commit", "-a", "-m", "moved some stuff"); err != nil {
+		t.Fatal(err)
+	}
+
+	// create the topic branch from the original branch point of the base branch
+	if _, err := runGit(ctx, ".", "checkout", "-b", t.Name(), baseBranch+"^"); err != nil {
+		t.Fatal(err)
+	}
+
+	// fully delete deleted
+	if err := os.RemoveAll(filepath.Clean("src/gtaintegration/deleted")); err != nil {
+		t.Fatal(err)
+	}
+
+	if testing.Verbose() {
+		out, err := runGit(ctx, ".", "status", "--short")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Logf("\n%s", out)
+	}
+
+	if _, err := runGit(ctx, ".", "commit", "-a", "-m", "delete some stuff"); err != nil {
+		t.Fatal(err)
+	}
+
+	// merge the base branch. After this, the branch topology should look like this:
+	// * (topic) merge
```

Comment:
Is this enough history to verify that this works better than `merge-base`? Like a commit to `master` and `topic` _after_ the merge?
```
* b91ae25 (HEAD -> main) add main
| * 075acfc (topic) add topic
| *   01c87c7 Merge branch 'main' into topic
| |\
| |/
|/|
* | 5220151 delete some stuff
| * 9dc58c8 moved some stuff
|/
* 67d81d6 initial commit
```

Reply:
It is enough, yes. The git differ used to use `base...branch`, and the tests demonstrate that was insufficient prior to the changes in the second commit.